src/client.ts

```diff
@@ -75,7 +75,7 @@ export class Client extends EventEmitter {
 
   constructor (options: IClientOptions) {
     super();
-    this.http = new HttpClient();
```

Comment:
When Streaming Client spins up a `HttpClient` instance, it now passes in the customHeaders from the config (though perhaps we should check if it exists first?) so that the `HttpClient` can then use it at a later point